### PR TITLE
Implementation of rule 3.5 - indentation

### LINKED ISSUE #62: Rule 3.5: Use spaces for indentation, indenting is equal to 4 spaces
**May be you can look if it is already done in ktlint-standard?** If yes - simply inherit it with customizations

Only spaces are allowed for indentation and each indentation should equal to 4 spaces (tabs are not allowed). 
In case you prefer using tabs - just simply configure auto change of tabs to spaces in your IDE.
Exceptions:
1) When breaking parameter list of a method/class constructor it can be aligned with 8 spaces or a parameter that was moved to a newline can be on the same level as the previous argument:
    
```kotlin
class Class(propertiesFileName: String,
            otherName: String) {
```

or
 
```kotlin
fun visit(
        node: ASTNode,
        autoCorrect: Boolean,
        params: KtLint.Params,
        emit: (offset: Int, errorMessage: String, canBeAutoCorrected: Boolean) -> Unit
) {
```
    
    
2) Operators like +/-/*/e.t.c can be indented with 8 spaces:
    
```kotlin
val abcdef = "my splitted" +
                " string"
```

### PR DESCRIPTION
**Pull request by petertrr**
This closes #62 

### What's done:
* Implemented rule
* Added tests
* Updated rules-config.json
* Updated available-rules.md

### Fixmes:
* [x] Handle additional spaces before `KDOC_LEADING_ASTERISK`s
* [x] Improve handling of line after binary operators
* [ ] Proper way to find top parent of chained dot calls
* [ ] Update code style guide according to implementation

### CODE REVIEW

File: diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/DiktatRuleSetProvider.kt [hunk 1]
```diff
@@ -39,6 +40,7 @@ class DiktatRuleSetProvider(private val jsonRulesConfig: String = "rules-config.
             IdentifierNaming(),
             BracesInConditionalsAndLoopsRule(),
             BlockStructureBraces(),
+            IndentationRule(),
```
Comment (akuleshov7) [inline diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/DiktatRuleSetProvider.kt:43]: we have discussed it, but I have forgotten - as I can remember we decided to put this rule to the end of the list, no?

Comment (petertrr) [reply diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/DiktatRuleSetProvider.kt:43]: Right, I will move it.

File: diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/files/IndentationRule.kt [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+    override fun visit(node: ASTNode, autoCorrect: Boolean, params: KtLint.Params, emit: (offset: Int, errorMessage: String, canBeAutoCorrected: Boolean) -> Unit) {
+        configRules = params.getDiktatConfigRules()
+        isFixMode = autoCorrect
+        emitWarn = emit
+        fileName = params.fileName!!
+
+        configuration = IndentationConfig(configRules.getRuleConfig(WRONG_INDENTATION)?.configuration
+                ?: mapOf())
+        customIndentationCheckers = listOf(
+                AssignmentOperatorChecker(configuration),
```
Comment (akuleshov7) [inline diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/files/IndentationRule.kt:71]: do you check lambdas?
a newline after ->

Comment (petertrr) [reply diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/files/IndentationRule.kt:71]: Not explicitly; every opening brace causes increase of indentation, and as a consequence body of lambda is indented.

**Pull request merged**